lib/aws-record/record/item_operations.rb

```diff
@@ -91,14 +94,20 @@ def assign_attributes(opts)
         end
       end
 
-      # Updates model attributes and then performs a conditional save
+      # Mass assigns the attributes to the model and then performs a save
+      # 
+      # @param [Hash] opts
       def update(opts)
         self.assign_attributes(opts)
 
         self.save
       end
 
-      # Updates model attributes and performs validation
+      # Updates model attributes and validates new values
+      #
+      # @param [Hash] opts
+      # @raise [Aws::Record::Errors::ValidationError] if any new values
+      #  violate the models validations.
       def update!(opts = {})
```

Comment:
Can `#update!` be called without any options? This seems to contradict what we did in `#update`. It possibly makes the most sense to require the hash to have some attributes.